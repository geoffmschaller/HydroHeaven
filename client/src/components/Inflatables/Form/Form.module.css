.form {
  width: 100%; }
  .form .input {
    width: 100%;
    position: relative;
    margin-bottom: 20px; }
    .form .input input {
      width: 100%;
      padding: 10px;
      border: 1px solid #abb2b9; }
    .form .input textarea {
      width: 100%;
      padding: 10px;
      border: 1px solid #abb2b9; }
    .form .input .label {
      position: absolute;
      top: 10px;
      left: 10px;
      -webkit-transition: all .2s ease-in-out;
      -o-transition: all .2s ease-in-out;
      transition: all .2s ease-in-out;
      background-color: white;
      font-size: 14px; }
    .form .input .open {
      top: -5px;
      left: 0;
      font-size: 10px;
      padding: 0 10px; }
  .form button {
    color: #f8c471;
    border: 2px solid #f8c471;
    float: right; }
    .form button i {
      -webkit-animation: Rotate 1s infinite;
      animation: Rotate 1s infinite; }
    .form button:hover {
      color: white;
      background-color: #f8c471; }
  .form .successButton {
    color: #7dcea0;
    text-align: right;
    font-size: 16px;
    font-weight: bold; }
  .form .errorMessage {
    color: red; }

@-webkit-keyframes Rotate {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg); }
  25% {
    -webkit-transform: rotate(90deg);
    transform: rotate(90deg); }
  50% {
    -webkit-transform: rotate(180deg);
    transform: rotate(180deg); }
  75% {
    -webkit-transform: rotate(270deg);
    transform: rotate(270deg); }
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg); } }

@keyframes Rotate {
  0% {
    -webkit-transform: rotate(0deg);
    transform: rotate(0deg); }
  25% {
    -webkit-transform: rotate(90deg);
    transform: rotate(90deg); }
  50% {
    -webkit-transform: rotate(180deg);
    transform: rotate(180deg); }
  75% {
    -webkit-transform: rotate(270deg);
    transform: rotate(270deg); }
  100% {
    -webkit-transform: rotate(360deg);
    transform: rotate(360deg); } }
