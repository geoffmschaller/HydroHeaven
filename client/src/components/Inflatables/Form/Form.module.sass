
.form
	width: 100%

	.input
		width: 100%
		position: relative
		margin-bottom: 20px

		input
			width: 100%
			padding: 10px
			border: 1px solid #abb2b9

		textarea
			width: 100%
			padding: 10px
			border: 1px solid #abb2b9

		.label
			position: absolute
			top: 10px
			left: 10px
			transition: all .2s ease-in-out
			background-color: white
			font-size: 14px

		.open
			top: -5px
			left: 0
			font-size: 10px
			padding: 0 10px

	button
		color: #f8c471
		border: 2px solid #f8c471
		float: right

		i
			animation: Rotate 1s infinite

		&:hover
			color: white
			background-color: #f8c471

	.successButton
		color: #7dcea0
		text-align: right
		font-size: 16px
		font-weight: bold

	.errorMessage
		color: red


@keyframes Rotate
	0%
		transform: rotate(0deg)

	25%
		transform: rotate(90deg)

	50%
		transform: rotate(180deg)

	75%
		transform: rotate(270deg)

	100%
		transform: rotate(360deg)