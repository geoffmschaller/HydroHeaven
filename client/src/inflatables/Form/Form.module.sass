@import "src/static/sass/config"

.form
	width: 100%

	input
		width: 100%
		padding: 10px
		border: 1px solid #abb2b9
		margin-bottom: 15px

	textarea
		width: 100%
		padding: 10px
		border: 1px solid #abb2b9
		margin-bottom: 15px

	.label
		position: absolute
		top: 10px
		left: 10px
		transition: all .2s ease-in-out
		background-color: white
		font-size: 14px

	button
		color: $primary
		border: 2px solid $primary
		background-color: transparent
		float: right

		i
			animation: Rotate 1s infinite

		&:hover
			color: white
			background-color: #f8c471

	.errorMessage
		color: #ec7063
		width: 100%
		text-align: center
		font-weight: bold

	.successMessage
		color: #5499c7
		width: 100%
		text-align: center
		font-weight: bold


@keyframes Rotate
	0%
		transform: rotate(0deg)

	25%
		transform: rotate(90deg)

	50%
		transform: rotate(180deg)

	75%
		transform: rotate(270deg)

	100%
		transform: rotate(360deg)