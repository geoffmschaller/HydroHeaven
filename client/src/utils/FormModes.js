export const FORM_READY = 'form_ready';
export const FORM_SUBMITTING = 'form_submitting';
export const FORM_SUCCESS = "form_sucess";