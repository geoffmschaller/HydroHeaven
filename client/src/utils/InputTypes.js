export const LINE_INPUT = 'line_input';
export const PASSWORD_INPUT = 'password_input';
export const MULTI_LINE_INPUT = 'multi_line_input';