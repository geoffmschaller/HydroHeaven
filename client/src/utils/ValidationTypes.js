export const EMAIL_VALIDATION = 'email_validation';
export const TEXT_VALIDATION = 'text_validation';