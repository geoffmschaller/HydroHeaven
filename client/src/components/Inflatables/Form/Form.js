import React from 'react';
import styles from './Form.module.css';
import PropTypes from 'prop-types';
import {FORM_READY, FORM_SUBMITTING, FORM_SUCCESS} from "../../../utils/FormModes";
import {ERROR, NONE, SUCCESS} from "../../../data/ActionResults";
import {LINE_INPUT, MULTI_LINE_INPUT, PASSWORD_INPUT} from "../../../utils/InputTypes";

class Form extends React.Component {

	render() {
		return (
			<div className={styles.form}>
				{
					this.props.inputs ? this.props.inputs.map((inp, index) => {

						if (inp.type === LINE_INPUT) {
							return <input type="text"
							              key={index}
							              placeholder={inp.name}
							              onChange={(event) => this.props.updateValue(event, index)}
							              value={inp.value}
							/>
						}
						if (inp.type === PASSWORD_INPUT) {
							return <input type="password"
							              key={index}
							              placeholder={inp.name}
							              onChange={(event) => this.props.updateValue(event, index)}
							              value={inp.value}
							/>
						}
						if (inp.type === MULTI_LINE_INPUT) {
							return <textarea placeholder={inp.name}
							                 key={index}
							                 onChange={(event) => this.props.updateValue(event, index)}
							                 value={inp.value}
							/>
						}
					}) : null
				}
				{
					this.props.mode === FORM_READY ?
						<button onClick={() => this.props.submit()}>Submit</button> : null
				}
				{
					this.props.mode === FORM_SUBMITTING ?
						<button><i className="fad fa-spinner-third"/></button> : null
				}
				<div className="clear"/>
				<div className={styles.errorMessage}>
					{
						this.props.messages && this.props.messages.status === ERROR ? <p>{this.props.messages.message}</p> : null
					}
				</div>
				<div className={styles.successMessage}>
					{
						this.props.messages && this.props.messages.status === SUCCESS ? <p>{this.props.messages.message}</p> : null
					}
				</div>
			</div>
		);
	}
}


Form.propTypes = {
	inputs: PropTypes.arrayOf(PropTypes.shape({
		name: PropTypes.string.isRequired,
		type: PropTypes.string.isRequired,
		required: PropTypes.bool.isRequired,
		value: PropTypes.string.isRequired
	})).isRequired,
	messages: PropTypes.shape({
		status: PropTypes.oneOf([SUCCESS, ERROR, NONE]).isRequired,
		message: PropTypes.string.isRequired
	}).isRequired,
	mode: PropTypes.oneOf([FORM_READY, FORM_SUBMITTING, FORM_SUCCESS]).isRequired,
	updateValue: PropTypes.func.isRequired,
	submit: PropTypes.func.isRequired
};

export default Form;