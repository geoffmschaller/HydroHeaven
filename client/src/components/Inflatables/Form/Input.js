import React from 'react';
import styles from "./Form.module.css";
import PropTypes from 'prop-types';
import {LINE_INPUT, MULTI_LINE_INPUT} from "../../../utils/InputTypes";

class Input extends React.Component {

	state = {
		open: false
	};

	checkLabel = (event, force = false) => {
		if (force) {
			this.setState({open: true})
		} else {
			let inp = event.target;
			if (inp.value === "") {
				this.setState({open: false})
			}
		}
	};

	render() {

		let labelClasses = this.state.open ? [styles.label, styles.open].join(" ") : styles.label;

		return (
			<div className={styles.input}>
				<label htmlFor="" className={labelClasses}>{this.props.inputData.name}:</label>

				{
					this.props.inputData.type === LINE_INPUT ?
						<input type="text"
						       onFocus={(event) => this.checkLabel(event, true)}
						       onBlur={(event) => this.checkLabel(event)}
						       onChange={this.props.change}
						       value={this.props.val}
						/> :
						<textarea onFocus={(event) => this.checkLabel(event, true)}
						          onBlur={(event) => this.checkLabel(event)}
						          onChange={this.props.change}
						          value={this.props.val}
						/>
				}

			</div>
		);
	}
}

Input.propTypes = {
	inputData: PropTypes.shape({
		name: PropTypes.string.isRequired,
		type: PropTypes.oneOf([LINE_INPUT, MULTI_LINE_INPUT]).isRequired,
		required: PropTypes.bool.isRequired,
		value: PropTypes.string.isRequired
	}).isRequired,
	change: PropTypes.func.isRequired,
	val: PropTypes.string.isRequired
};

export default Input;