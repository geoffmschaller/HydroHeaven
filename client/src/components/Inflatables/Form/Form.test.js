import React from 'react';
import Form from "./Form";
import {FORM_READY} from "../../../utils/FormModes";
import {LINE_INPUT} from "../../../utils/InputTypes";
import {TEXT_VALIDATION} from "../../../utils/ValidationTypes";

const formData = {
	mode: FORM_READY,
	result: null,
	url: "/api/customer-contact/",
	inputs: [
		{
			name: "Name",
			type: LINE_INPUT,
			required: true,
			value: "",
			validator: TEXT_VALIDATION
		}
	]
};
const wrapper = <Form formData={formData}/>;

it("Form Inflatable", () => {
	expect(wrapper).toMatchSnapshot();
});