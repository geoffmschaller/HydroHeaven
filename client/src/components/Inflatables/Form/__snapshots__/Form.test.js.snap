// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Inflatable 1`] = `
<Form
  formData={
    Object {
      "inputs": Array [
        Object {
          "name": "Name",
          "required": true,
          "type": "line_input",
          "validator": "text_validation",
          "value": "",
        },
      ],
      "mode": "form_ready",
      "result": null,
      "url": "/api/customer-contact/",
    }
  }
/>
`;
